contracts/GeneratedExperience.cdc

```diff
@@ -0,0 +1,350 @@
+		pub let season: UInt64
+		pub let royalties: [FindPack.Royalty]
+		pub let squareImage: String
+		pub let bannerImage: String
+		pub let description: String
+		access(self) let extra: {String: AnyStruct}
+
+		init(
+			season: UInt64,
+			royalties: [FindPack.Royalty],
+			squareImage: String,
+			bannerImage: String,
+			description: String
+		) {
+			self.season = season
+			self.royalties = royalties
+			self.squareImage = squareImage
+			self.bannerImage = bannerImage
+			self.description = description
+			self.extra={}
+		}
+	}
+
+	pub struct Info {
+		pub let season: UInt64
+		pub let name: String
+		pub let description: String
+		pub let thumbnailHash: String
+		pub let fullsizeHash: String
+		pub let edition: UInt64
+		pub let maxEdition: UInt64
+		pub let artist: String
+		pub let rarity: String
+		access(self) let extra: {String: AnyStruct}
+
+		init(season: UInt64, name: String, description: String, thumbnailHash: String, edition:UInt64, maxEdition:UInt64, fullsizeHash: String, artist: String, rarity: String) {
+			self.season=season
+			self.name=name
+			self.description=description
+			self.thumbnailHash=thumbnailHash
+			self.edition=edition
+			self.maxEdition=maxEdition
+			self.fullsizeHash=fullsizeHash
+			self.artist=artist
+			self.rarity=rarity
+			self.extra={}
+		}
+	}
+
+	pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
+		pub let id: UInt64
+		pub let info: Info
+
+		init(
+			info: Info
+		) {
+			self.id = self.uuid
+			self.info=info
+		}
+
+		pub fun getViews(): [Type] {
+			return [
+			Type<MetadataViews.Display>(),
+			Type<MetadataViews.Royalties>(),
+			Type<MetadataViews.Editions>(),
+			Type<MetadataViews.Traits>(),
+			Type<MetadataViews.ExternalURL>(),
+			Type<MetadataViews.NFTCollectionData>(),
+			Type<MetadataViews.NFTCollectionDisplay>(),
+			Type<MetadataViews.Medias>(),
+			Type<MetadataViews.Rarity>(),
+			Type<FindPack.PackRevealData>()
+			]
+		}
+
+		pub fun resolveView(_ view: Type): AnyStruct? {
+
+			let collection = GeneratedExperience.collectionInfo[self.info.season]!
+			let imageFile = MetadataViews.IPFSFile( cid: self.info.thumbnailHash, path: nil)
+
+			switch view {
+
+			case Type<FindPack.PackRevealData>():
+				let data : {String : String} = {
+					"nftImage" : imageFile.uri() ,
+					"nftName" : self.info.name,
+					"packType" : "GeneratedExperience"
+				}
+				return FindPack.PackRevealData(data)
+
+			case Type<MetadataViews.Display>():
+				return MetadataViews.Display(
+					name: self.info.name,
+					description: self.info.description,
+					thumbnail: MetadataViews.IPFSFile(
+						cid: self.info.thumbnailHash, path: nil
+					)
+				)
+			case Type<MetadataViews.Editions>():
+				// We do not show season here unless there are more than 1 collectionInfo (that is indexed by season)
+				var editionName = "edition"
+				if GeneratedExperience.collectionInfo.length > 1 {
+					editionName = "season ".concat(self.info.season.toString()).concat(" edition")
```

Comment:
hmm i am not sure what this edition name should be here. 

can we talk about this?

I really want to be able to have a filter that says 

"give me all generated experiences from all seasons that are below edition 1"

in order to do this in typesense the name of the edition have to be the same. So i almost think that we want one edition with the name "experience" or something, or even just leave it unnamed. 

Multiple editions are very strange on flow atm I think.

Reply:
genereatedexperience